foursight_core/app_utils.py

```diff
@@ -334,6 +574,191 @@ def trim_output(cls, output, max_size=100000):
+        in a form that looks like 2022-08-22 13:25:34 EDT.
+
+        :param time_t: Epoch time value (i.e. seconds since 1970-01-01T00:00:00Z)
+        :return: US/Eastern datetime string (e.g.: 2022-08-22 13:25:34 EDT).
+        """
+        try:
+            if not isinstance(time_t, int):
+                return ""
+            t = datetime.datetime.fromtimestamp(time_t, tz=pytz.UTC)
+            return self.convert_utc_datetime_to_useastern_datetime(t)
+        except Exception as e:
+            self.note_non_fatal_error_for_ui_info(e, 'convert_time_t_to_useastern_datetime')
+            return ""
+
+    def ping_elasticsearch(self, env_name: str) -> bool:
+        logger.warn(f"foursight_core: Pinging ElasticSearch: {self.host}")
+        try:
+            response = self.init_connection(env_name).connections["es"].test_connection()
+            logger.warn(f"foursight_core: Done pinging ElasticSearch: {self.host}")
+            return response
+        except Exception as e:
+            logger.warn(f"Exception pinging ElasticSearch ({self.host}): {e}")
+            return False
+
+    def ping_portal(self, env_name: str) -> bool:
+        portal_url = ""
+        try:
+            portal_url = self.get_portal_url(env_name)
+            logger.warn(f"foursight_core: Pinging portal: {portal_url}")
+            response = requests.get(portal_url, timeout=4)
+            logger.warn(f"foursight_core: Done pinging portal: {portal_url}")
+            return (response.status_code == 200)
+        except Exception as e:
+            logger.warn(f"foursight_core: Exception pinging portal ({portal_url}): {e}")
+            return False
+
+    def ping_sqs(self) -> bool:
+        sqs_url = ""
+        try:
+            sqs_url = self.sqs.get_sqs_queue().url
+            logger.warn(f"foursight_core: Pinging SQS: {sqs_url}")
+            sqs_attributes = self.sqs.get_sqs_attributes(sqs_url)
+            logger.warn(f"foursight_core: Done pinging SQS: {sqs_url}")
+            return (sqs_attributes is not None)
+        except Exception as e:
+            logger.warn(f"Exception pinging SQS ({sqs_url}): {e}")
+            return False
+
+    def reload_lambda(self, lambda_name: str = None) -> bool:
+        """
+        Experimental.
+        Reloads the lambda code for the given lambda name. We do this by making an innocuous change
+        to it, namely, by adding/removing a trailing dot to its description. This causes the lambda
+        to be reloaded, however this also changes its last modified date, which we would also like to
+        to accurately get (get_lambda_last_modified), so we store its original value in a lambda tag,
+        the updating of which does not update the modified time; so the code (get_lambda_last_modified)
+        to get the last modified time of the lambda needs to look first at this tag and take that
+        if it exists before looking at the real last modified time.
```

Comment:
That is a neat trick - what do you use it for? For reloading the GAC instantaneously?

Reply:
Yes I had a need to observe the behavior when the app is reloaded, AWS periodically does because it unloads after some time of activity; and also had a need because (I forget precisely but) there was some issue with AWS permissions which I was experimenting with fixing, to see if fixing the permission fixed the issue but could not do this unless/until the app was reloaded and I didn't want to want to the AWS auto-unload/reload.